job.finished: fix returnvalue already parsed and returnvalue in watchdog

saviogrossi opened the pull request:
more fixes in job.finished returns.

manast commented on lib/job.js:
```diff
@@ -7,7 +7,7 @@ var scripts = require('./scripts');
 
-var FINISHED_WATCHDOG = 5000;
+var FINISHED_WATCHDOG = 1000;
 
```
I guess the best would be to have the watchdog timeout as a configurable setting.

amirrustam replied:
Making this configurable would be helpful.

manast commented on lib/job.js:
```diff
@@ -7,7 +7,7 @@ var scripts = require('./scripts');
 
-var FINISHED_WATCHDOG = 5000;
+var FINISHED_WATCHDOG = 1000;
 
```
(I can fix this later on)

manast commented on lib/job.js:
```diff
@@ -368,14 +368,14 @@ Job.prototype.finished = function(){
-            var finished = status > 0;
-            if(finished){
+            Job.fromId(_this.queue, _this.id).then(function(job){
+              removeListeners();
```
I am thinking that maybe it is better to call removeListeners() on a .finally() call on ```scripts.isFinished```, and return Job.fromId so that the errors propagate correctly. A catch with a ```console.error()``` would also be nice :).

manast commented on lib/job.js:
```diff
@@ -368,14 +368,14 @@ Job.prototype.finished = function(){
-              removeListeners();
-              resolve();
-            }
+            });
```
I do not think this code is correct. It is ignoring the status result from the call to isFinished, so it will resolve independently of if the job has been finished or not.

saviogrossi replied:
Ok, I will work on this

saviogrossi replied:
Sorry, I misunderstood isFinished, I thought it only resolve when job status is already finished. Will fix this, tks.

strax commented:
@saviogrossi @manast what's the status with this? I could submit a PR with only the return value fix if you're interested as we're already using a custom fork with that patch in.

manast commented:
@strax there are un-addressed comments yet...

saviogrossi commented:
sorry @manast <https://github.com/manast> @strax
<https://github.com/strax> , I
haven't much time to address Manuel comments. Sami, if you think you can
address, I will cancel this PR so you send yours. Tks

On Wed, Sep 27, 2017 at 8:45 AM, Manuel Astudillo <notifications@github.com>
wrote:

> @strax <https://github.com/strax> there are un-addressed comments yet...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OptimalBits/bull/pull/622#issuecomment-332495288>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAqcAHHlhsme2PpypY162rhqLCZ6azNmks5smjV1gaJpZM4OdRTC>
> .
>
